core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckService.java

```diff
@@ -395,6 +401,14 @@ public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exc
 
         return HttpResponse.from(updateHandler.handle(ctx, req).thenApply(updateResult -> {
             if (updateResult != null) {
+                for (HealthCheckUpdateListener updateListener : updateListeners) {
+                    try {
+                        updateListener.onUpdate(updateResult);
+                    } catch (Throwable t) {
+                        logger.warn("An error occurred when notifying a Update event", t);
```

Comment:
`Unexpected exception from HealthCheckUpdateListener.onUpdate():` ?

Reply:
And how about notifying after calling `setHealthy()`, so that the notification can potentially be made sooner?